python/pip_install/pip_repository.bzl

```diff
@@ -349,100 +349,74 @@ alias(
+        macro_tmpl = "@{name}//{{}}".format(name = rctx.attr.name)
+        all_whl_reqs_tmpl = "@{name}//{{}}:whl".format(name = rctx.attr.name)
+    else:
+        macro_tmpl = "@{name}_{{}}//:pkg".format(name = rctx.attr.name)
+        all_whl_reqs_tmpl = "@{name}_{{}}//:whl".format(name = rctx.attr.name)
+
+    all_requirements = []
+    all_whl_requirements = []
+    if bzlmod:
+        for p in pkgs.bzl_packages:
+            all_requirements.append(macro_tmpl.format(p, p) if incompat_aliases else macro_tmpl.format(p, "pkg"))
+            all_whl_requirements.append(macro_tmpl.format(p, "whl"))
+    else:
+        for p in pkgs.bzl_packages:
+            all_requirements.append(macro_tmpl.format(p))
+            all_whl_requirements.append(all_whl_reqs_tmpl.format(p))
 
-    rctx.file("BUILD.bazel", build_contents)
-    rctx.template("requirements.bzl", rctx.attr._template, substitutions = {
-        "%%ALL_REQUIREMENTS%%": _format_repr_list([
-            macro_tmpl.format(p, p) if rctx.attr.incompatible_generate_aliases else macro_tmpl.format(p, "pkg")
-            for p in bzl_packages
-        ]),
-        "%%ALL_WHL_REQUIREMENTS%%": _format_repr_list([
-            macro_tmpl.format(p, "whl")
-            for p in bzl_packages
-        ]),
-        "%%MACRO_TMPL%%": macro_tmpl,
-        "%%NAME%%": rctx.attr.name,
-        "%%REQUIREMENTS_LOCK%%": str(requirements_txt),
-    })
-
-pip_repository_bzlmod_attrs = {
-    "incompatible_generate_aliases": attr.bool(
-        default = False,
-        doc = "Allow generating aliases in '@pip//:<pkg>' -> '@pip_<pkg>//:pkg'. This replaces the aliases generated by the `bzlmod` tooling.",
-    ),
-    "repo_name": attr.string(
-        mandatory = True,
-        doc = "The apparent name of the repo. This is needed because in bzlmod, the name attribute becomes the canonical name",
-    ),
-    "requirements_darwin": attr.label(
-        allow_single_file = True,
-        doc = "Override the requirements_lock attribute when the host platform is Mac OS",
-    ),
-    "requirements_linux": attr.label(
-        allow_single_file = True,
-        doc = "Override the requirements_lock attribute when the host platform is Linux",
-    ),
-    "requirements_lock": attr.label(
-        allow_single_file = True,
-        doc = """
-A fully resolved 'requirements.txt' pip requirement file containing the transitive set of your dependencies. If this file is passed instead
-of 'requirements' no resolve will take place and pip_repository will create individual repositories for each of your dependencies so that
-wheels are fetched/built only for the targets specified by 'build/run/test'.
-""",
-    ),
-    "requirements_windows": attr.label(
-        allow_single_file = True,
-        doc = "Override the requirements_lock attribute when the host platform is Windows",
-    ),
-    "_template": attr.label(
-        default = ":pip_repository_requirements_bzlmod.bzl.tmpl",
```

Comment:
What is more, since there are subtle differences, it is easy to break one or the other during refactorings. Having separate templates ensures that we are explicit with what we support.